documentation/src/main/asciidoc/topics/search.adoc

```diff
@@ -0,0 +1,1087 @@
+[id='search_api']
+=  Indexing and Searching
+
+== Overview
+
+{brandname} provides a search API that lets you index and search cache values stored as Java POJOs or as objects encoded as link:https://developers.google.com/protocol-buffers/[Protocol Buffers].
+
+Searching is possible both in link:#query_library[library] and link:#query_remote[client/server mode] (for Java, C#, Node.js and other clients), and {brandname} can index data using link:http://lucene.apache.org/[Apache Lucene], offering an efficient link:https://en.wikipedia.org/wiki/Full-text_search[full-text] capable search engine in order to cover a wide range of data retrieval use cases.
+
+Indexing configuration relies on a schema definition, and for that {brandname} can use annotated Java classes when in library mode, and protobuf schemas for remote clients. By standardizing on protobuf, {brandname} allows full interoperability between Java and non-Java clients.
+
+{brandname} has its own query language called _link:#query_ickle[Ickle]_, which is string-based and adds support for full-text searching. Ickle support searches over indexed data, partially indexed
+data or non-indexed data.
+
+Finally, {brandname} has support for link:#query_continuous[Continuous Queries], which works in a reverse manner to the other APIs: instead of creating, executing a query
+and obtain results, it allows a client to register queries that will be evaluated continuously as data in the cluster changes, generating notifications
+whenever the changed data matches the queries.
+
+== Indexing Entry Values
+
+Indexing entry values in {brandname} caches dramatically improves search performance and allows you to perform full-text queries. However, indexing can degrade write throughput for {brandname} clusters. For this reason you should plan to use strategies to optimise query performance, depending on the cache mode and your use case. More information on link:#query_performance[query performance guide].
+
+=== Configuration
+
+To enable indexing via XML, you need to add the `<indexing>` element to your cache configuration, specify the entities that are indexed and optionally pass additional properties.
+
+[NOTE]
+====
+The presence of an `<indexing>` element which omits the `enabled` attribute will auto-enable indexing for your
+convenience, even though the default value of the `enabled` attribute is defined as `"false"` in the XSD schema.
+This behaviour exists only on schemas starting with version 11.
```

Comment:
I'd drop "This behaviour exists only on schemas starting with version 11." Hardcoding the version doesn't work downstream. Also, I'm not sure it adds value here.

I seem to recall this but maybe it was somewhere else. Why don't we change the default to "true" in the XSD?

Reply:
it has to do with some weirdness in the configuration, I can't remember what it was (@anistor) worked on it. I will just keep the note and remove the hardcoded version